blueocean-dashboard/src/main/js/redux/actions.js

```diff
@@ -757,9 +766,37 @@ export const actions = {
                     logUrl,
                 },
             }))
-            .catch((error) => console.error('error', error));
+            .catch((error) => console.error('error', error)); // eslint-disable-line no-console
             }
             return null;
         };
     },
+    
+    fetchTestResults(config, runDetails) {
+        return (dispatch) => {
+            // const appUrlBase = config.getAppURLBase(); // TODO: Should use this instead
+            const appUrlBase = '/jenkins';
+            let url;
+            
+            if (runDetails.isMultiBranch) {
+                url = 'TODO'; // Construct different URL for multi-branch
+            } else {
+                // eslint-disable-next-line max-len
+                url = `${appUrlBase}/blue/rest/organizations/${runDetails.organization}/pipelines/${runDetails.branch}/runs/${runDetails.runId}/testReport/result`;
+            }
```

Comment:
Check out the `buildRunDetailsUrl` function in `UrlUtils` for a utility to help build this URL so that it is fully escaped.